addons/mail/static/src/core/common/message_search_hook.js

```diff
@@ -0,0 +1,90 @@
+/* @odoo-module */
+
+import { useSequential } from "@mail/utils/common/hooks";
+import { useState, onWillUnmount, markup } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+
+/**
+ * @param {string} searchTerm
+ * @param {string} target
+ */
+export function searchHighlight(searchTerm, target) {
+    if (!searchTerm) {
+        return target;
+    }
+    const htmlDoc = new DOMParser().parseFromString(target, "text/html");
+    for (const term of searchTerm.split(" ")) {
+        const matchs = htmlDoc.evaluate(
+            `//*[text()[contains(translate(., '${term.toUpperCase()}', '${term.toLowerCase()}'), '${term.toLowerCase()}')]]`,
```

Comment:
I'm not sure if it's related to the previous comment about escape, but when posting `<strong>test</strong>` (as text) and then searching for it, I get no result.

I guess it's because tags are escaped in DB, but this is confusing.
Maybe search term should be html escaped before looking into DB as well, so that `<strong>test</strong>`  becomes `&lt;strong&gt;test&lt;/strong&gt;` which gives a result?

There should be a test for that.